.github/workflows/rust.yml

```diff
@@ -10,80 +15,79 @@ jobs:
       matrix:
         rust:
           - stable
-          - 1.44.0
+          - 1.63.0
           - nightly
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v4
+      - name: Install ${{ matrix.rust }} toolchain
+        uses: dtolnay/rust-toolchain@master
         with:
-          profile: minimal
           toolchain: ${{ matrix.rust }}
-          override: true
       - name: Cargo update
         run: cargo update
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
+      - run: RUSTFLAGS="-D warnings" cargo check
 
   test:
     name: Test Suite
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
+      - uses: actions/checkout@v4
+      - name: Install stable toolchain
+        uses: dtolnay/rust-toolchain@stable
+      - run: cargo test
```

Comment:
Note that for other crates of rusticata, a matrix of tests will probably be required for different feature sets (at least `default` and `all-features`
`no-std` would also be nice to have in the feature test matrix, but unfortunately I already know that most of the crates will fail (tests are not annotated for `std`/`no-std` at the moment)

Reply:
:ballot_box_with_check: Heard. I will make sure that for crates with features we maintain coverage in the matrix.

Another idea: in the Rustls org we've been using [cargo hack](https://github.com/taiki-e/cargo-hack) to test the feature powerset automatically. Depending on the number of features this can end up being combinatorically pretty expensive - in those cases we've been lifting the test into a separate workflow that runs once a day instead of for every PR. For a crate with 2-3 features it fits well into the main workflow.

If you're open to the idea I could explore this option when I update the first crate with some feature flags.

Reply:
I hear you on not wanting to depend on tools that might go away. I'll give this some thought and we can think about it on a case-by-case basis as we update repos with features. Thanks!